src/App.jsx

```diff
@@ -14,33 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getGoodsPrepared(goods, goodsSortBy, reverse) {
+  let goodsPrepared = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (goodsSortBy) {
+    goodsPrepared = goodsPrepared.sort((good1, good2) => {
+      switch (goodsSortBy) {
+        case 'abc':
+          return good1 > good2 ? 1 : -1;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case 'length':
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    goodsPrepared = goodsPrepared.reverse();
+  }
+
+  return goodsPrepared;
+}
+
+// let COUNTER_PLUS = 1;
+
+export const App = () => {
+  const [goodsSortBy, setGoodsSortBy] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const renderingGoods = getGoodsPrepared(
+    goodsFromServer,
+    goodsSortBy,
+    reverse,
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': goodsSortBy === '' || goodsSortBy === 'length',
+          })}
+          onClick={() => {
+            setGoodsSortBy('abc');
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': goodsSortBy === '' || goodsSortBy === 'abc',
+          })}
+          onClick={() => {
+            setGoodsSortBy('length');
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': reverse === false,
+          })}
+          onClick={() => {
+            setReverse(!reverse);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(goodsSortBy || reverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setReverse(false);
+              setGoodsSortBy('');
+            }}
```

Comment:
```suggestion
            onClick={reset}
```
Consider moving this function from jsx